test/mailers/access_request_mailer_test.rb

```diff
@@ -24,37 +22,58 @@
       ENV['REQUEST_ACCESS_EMAIL_PREFIX'] = @original_prefix
     end
 
-    it 'is from deploys@' do
-      subject.from.must_equal ['deploys@samson-deployment.com']
-    end
+    describe 'multiple projects' do
+      before do
+        Project.any_instance.stubs(:valid_repository_url).returns(true)
+        Project.create!(name: 'Second project', repository_url: 'git://foo.com:hello/world.git')
+        AccessRequestMailer.access_request_email(
+          hostname, user, manager_email, reason, Project.all.map(&:id)).deliver_now
+      end
 
+      it 'is from deploys@' do
+        subject.from.must_equal ['deploys@samson-deployment.com']
+      end
 
-    it 'sends to configured addresses' do
-      subject.to.must_equal(address_list.split << manager_email)
-    end
+      it 'sends to configured addresses' do
+        subject.to.must_equal(address_list.split << manager_email)
+      end
 
-    it 'includes name in subject' do
-      subject.subject.must_match /#{user.name}/
-    end
+      it 'includes name in subject' do
+        subject.subject.must_match /#{user.name}/
+      end
 
-    it 'includes proper role in subject' do
-      subject.subject.must_match /#{Role.find(user.role_id + 1).name}/
-    end
+      it 'includes proper role in subject' do
+        subject.subject.must_match /#{Role.find(user.role_id + 1).name}/
+      end
 
-    it 'includes email in body' do
-      subject.body.to_s.must_match /#{user.email}/
-    end
+      it 'includes email in body' do
+        subject.body.to_s.must_match /#{user.email}/
+      end
 
-    it 'includes proper role in body' do
-      subject.body.to_s.must_match /#{Role.find(user.role_id + 1).name}/
-    end
+      it 'includes proper role in body' do
+        subject.body.to_s.must_match /#{Role.find(user.role_id + 1).name}/
+      end
+
+      it 'includes host in body' do
+        subject.body.to_s.must_match /#{hostname}/
+      end
 
-    it 'includes host in body' do
-      subject.body.to_s.must_match /#{hostname}/
+      it 'includes reason in body' do
+        subject.body.to_s.must_match /#{reason}/
+      end
+
+      it 'includes target project names in body' do
+        Project.all.each { |project| subject.body.to_s.must_match /#{project.name}/ }
+      end
     end
 
-    it 'includes reason in body' do
-      subject.body.to_s.must_match /#{reason}/
+    describe 'single project' do
+      before { AccessRequestMailer.access_request_email(
+          hostname, user, manager_email, reason, [projects(:test).id]).deliver_now }
+
+      it 'includes target project name in body' do
+        subject.body.to_s.must_match /#{projects(:test).name}/
+      end
```

Comment:
FWIW, I don't really believe in being strict about "have a separate test for everything" philosophy.  Well, I would be more in favor of that, if it weren't for the fact that each test is making at least one database insert.  This adds up and slows down the test suite.

I'd probably do this by having one test each for the "to" address, the subject, and the body, and more than one assertion in each test.